Restic documentation - Backup (#144)
backube/scribe: docs/usage/restic/index.rst

=======================
Restic Database Example
=======================

Restic
------
`Restic <https://restic.readthedocs.io/>`_ is a fast and secure backup program. 

> The following example will use the Restic backup and take a Snapshot at the source.

Reviewer: ```suggestion
The following example will use Restic to create a backup of a source volume.
```

   STORAGECLASS	  AGE
   persistentvolumeclaim/mysql-pv-claim   Bound     pvc-adbf57f1-6399-4738-87c9-4c660d982a0f   2Gi        RWO
   csi-hostpath-sc   60m

Add a new database.

.. code:: bash

   kubectl exec --stdin --tty -n source `kubectl get pods -n source | grep mysql | awk '{print $1}'` -- /bin/bash
   mysql -u root -p$MYSQL_ROOT_PASSWORD
   > show databases;
   +--------------------+
   | Database           |
   +--------------------+
   | information_schema |
   | mysql              |
   | performance_schema |
   | sys                |
   +--------------------+
   4 rows in set (0.00 sec)

   > create database synced;
   > exit
   exit
   
Restic Repository Setup
-----------------------

For the purpose of this tutorial we are using minio as S3 storage.
Start ``minio`` 

.. code:: bash

   hack/run-minio.sh 

   kubectl port-forward --namespace minio svc/minio 9000:9000

Once ``minio`` is up and running, create a restic repository for storing backup snapshots.
The contents of a directory at a specific point in time is called a "snapshot" in restic.

.. code:: bash

   export AWS_ACCESS_KEY_ID=access
   export AWS_SECRET_ACCESS_KEY=password

>    restic -r s3:http://127.0.0.1:9000/restic-repo init

Reviewer: The user shouldn't need to manually init the repo... the mover should do it on 1st sync.

On password prompt use the same password as used in setting up the ``restic-config``.
It should match the field ``RESTIC_PASSWORD`` in the secret. 

> Now, deploy the ``restic-config`` followed by ``ReplicationSource`` configuration.

Reviewer: Please include commented copies of the yamls here.
